lib/models/node.js

```diff
@@ -64,6 +69,31 @@ function NodeModelFactory (Model, waterline, _) {
                     return !_.isEmpty(catalog);
                 });
             }
+        },
+        addTags: function(id, tags) {
+            return Promise.fromNode(this.native.bind(this))
+                .then(function(collection) {
+                    return collection.update(
+                        { _id: waterline.nodes.mongo.objectId(id) },
+                        { 
+                            $addToSet: { tags: { $each: tags } },
+                            $set: { updatedAt: new Date() }
+                        });
+                });
+        },
+        remTags: function(id, tag) {
+            return Promise.fromNode(this.native.bind(this))
+                .then(function(collection) {
+                    return collection.update(
```

Comment:
@anhou yep, @tldavies is also going to do the same thing for the findAndModify atomic operation you suggested a while back for lookup entries on-dhcp-proxy I believe.